### PR TITLE
[5.1] Convert mod_wrapper to service provider

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Converts the wrapper module to service provider.

### Testing Instructions

1. Make sure you have site module mod_wrapper published and configured
2. You should see module output on frontend

mod_wrapper

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: modules/mod_wrapper/tmpl/default.php [hunk 1]
```diff
@@ -4,11 +4,11 @@
  * @package     Joomla.Site
  * @subpackage  mod_wrapper
  *
- * @copyright   (C) 2006 Open Source Matters, Inc. <https://www.joomla.org>
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (sandewt) [inline modules/mod_wrapper/tmpl/default.php]: 2006 is correct, the date 2024 is incorrect

File: modules/mod_wrapper/tmpl/default.php [hunk 2]
```diff
@@ -4,11 +4,11 @@
...
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
-defined('_JEXEC') or die;
+\defined('_JEXEC') or die;
```
Comment (sandewt) [inline modules/mod_wrapper/tmpl/default.php]: skip this change
`defined` is correct, `\defined` is not correct

**Comment by sandewt** [comment]
https://github.com/joomla/joomla-cms/blob/c74906ea80039cb1c01093648b3dc310d2807cc4/modules/mod_wrapper/src/Helper/WrapperHelper.php#L35

`public static function getParams(&$params)` is deprecated  will be removed in 6.0
Please change this function to a non-static function.

File: modules/mod_wrapper/tmpl/default.php (hunk 1 above)
Comment (joomlaweby) [reply modules/mod_wrapper/tmpl/default.php]: fixed

File: modules/mod_wrapper/tmpl/default.php (hunk 2 above)
Comment (joomlaweby) [reply modules/mod_wrapper/tmpl/default.php]: fixed

**Comment by joomlaweby** [comment]
> `public static function getParams(&$params)` is deprecated will be removed in 6.0 Please change this function to a non-static function.

@sandewt  resolved..thanks for review

File: modules/mod_wrapper/src/Helper/WrapperHelper.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ class WrapperHelper
      *
      * @since   1.5
      */
-    public static function getParams(&$params)
+    public function getParams(&$params)
```
Comment (sandewt) [inline modules/mod_wrapper/src/Helper/WrapperHelper.php]: Well done, but there is still work to be done.

Give the non static function a new name, eg:
```suggestion
    public function getParamsWrappper(Registry $params, SiteApplication $app)
```
Indicate the static function as deprecated:
`public static function getParams(&$params)`

Examples can be found in the already converted modules.

File: modules/mod_wrapper/src/Helper/WrapperHelper.php [hunk 4]
```diff
@@ -4,7 +4,7 @@
  * @package     Joomla.Site
  * @subpackage  mod_wrapper
  *
- * @copyright   (C) 2006 Open Source Matters, Inc. <https://www.joomla.org>
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (sandewt) [inline modules/mod_wrapper/src/Helper/WrapperHelper.php]:
```suggestion
 * @copyright   (C) 2006 Open Source Matters, Inc. <https://www.joomla.org>
```

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 62151be2f7df16162eb50790192243cdc35d2ccd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42792">issues.joomla.org/tracker/joomla-cms/42792</a>.</sub>

File: modules/mod_wrapper/src/Helper/WrapperHelper.php (hunk 4 above)
Comment (joomlaweby) [reply modules/mod_wrapper/src/Helper/WrapperHelper.php]: Fixed

File: modules/mod_wrapper/src/Helper/WrapperHelper.php (hunk 3 above)
Comment (joomlaweby) [reply modules/mod_wrapper/src/Helper/WrapperHelper.php]: Resolved...thanks for hints. I just do not know, why build failed now

File: modules/mod_wrapper/src/Helper/WrapperHelper.php [hunk 5]
```diff
@@ -66,4 +69,24 @@ public static function getParams(&$params)
...
+    /**
+     * Gets the parameters for the wrapper
+     *
+     * @param   mixed  &$params  The parameters set in the administrator section
+     *
+     * @return  mixed  &$params  The modified parameters
+     *
+     * @since   1.5
+     *
+     * @deprecated 5.1 will be removed in 6.0
```
Comment (sandewt) [inline modules/mod_wrapper/src/Helper/WrapperHelper.php]:
```suggestion
     * @deprecated __DEPLOY_VERSION__ will be removed in 6.0
```

Comment (joomlaweby) [reply modules/mod_wrapper/src/Helper/WrapperHelper.php]: Fixed

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 4706e0bceb53353d5b2d9a0ff0d6b68f2d661555<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42792">issues.joomla.org/tracker/joomla-cms/42792</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 4706e0bceb53353d5b2d9a0ff0d6b68f2d661555<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42792">issues.joomla.org/tracker/joomla-cms/42792</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42792">issues.joomla.org/tracker/joomla-cms/42792</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @joomlaweby  and also testing and review @sandewt @fgsw @viocassel @ceford

**Pull request merged**